cni/network/plugin/main.go

```diff
@@ -133,6 +133,52 @@ func handleIfCniUpdate(update func(*skel.CmdArgs) error) (bool, error) {
 	return isupdate, nil
 }
 
+// startTelemetryService - Kills if any telemetry service runs and start new telemetry service
+func startTelemetryService(path string) error {
+	platform.KillProcessByName(telemetry.TelemetryServiceProcessName)
+
+	log.Printf("[cni] Starting telemetry service process")
+
+	if err := common.StartProcess(path); err != nil {
+		log.Printf("[Telemetry] Failed to start telemetry service process :%v", err)
+		return err
+	}
+
+	log.Printf("[cni] Telemetry service started")
+
+	for attempt := 0; attempt < 5; attempt++ {
+		if telemetry.CheckIfSockExists() {
```

Comment:
telemetry.SockExists() ?

Why are there 5 attempts?

Reply:
just a calculated number. wait for second and then exit if telemetry socket doesn;t exist.